### PR TITLE
Support markers at start and midpoint of edges

### LINKED ISSUE #194: Arrows for both sides?
Is here http://cpettitt.github.io/project/dagre-d3/latest/demo/arrows.html a entry for an arrow type "both"? Arrows on both sides? I dont want two lines. I want set a single "edge":
g.setEdge(from, to, { arrowhead: both });

Is this possible? ;)

### PR DESCRIPTION
**Pull request by AlanHohn**
Adds "arrowtail" and "arrowmid" attributes to edges. 

Default to "undirected" so default behavior is the same.

Fixes #194

### CODE REVIEW

**Comment by lutzroeder** [comment]
Can you resolve conflicts and maybe also a short sample how this is being used?

**Comment by AlanHohn** [comment]
Thanks for taking a look. I fixed the merge conflict.

I did make some changes to `demos/arrows.html` to show how you'd use it; please let me know if you prefer me to add samples somewhere else as well.

Basically it adds the attributes `arrowmid` and `arrowtail` to edges so you can place markers there as well. The value of those attributes is identical to `arrowhead` (i.e. the name of the arrow style).

I did have to do a little update to the standard "normal" and "vee" arrows to detect when they're being used as `arrowtail` so they can draw themselves upside down (`orient: auto` on the marker aligns them with the edge but it doesn't flip the point of the arrow). Anyone who wants to use `arrowtail` with custom arrows would need to do the same but the change doesn't break any existing use of `arrowhead`.

So long story short you'd make a double-headed arrow edge like this:

```javascript
g.setEdge("leftNode", "rightNode", {
  arrowhead: "normal",
  arrowtail: "normal"
});
```

**Comment by fanxiaolong** [comment]
I need this feature,thx!

**Comment by mhyleomak** [comment]
Hi all,
Will this feature branch merge to master? It would be a great feature for me. 
Thanks!

**Comment by clement911** [comment]
I'd love to use this too!

**Comment by ralphtq** [comment]
Also keen to have this merge

**Comment by ralphtq** [comment]
Did the merge happen?

**Comment by lutzroeder** [comment]
Updated to #361.
